dandi/__init__.py

```diff
@@ -36,3 +36,57 @@ def set_logger_level(lgr, level):
 set_logger_level(lgr, os.environ.get("DANDI_LOG_LEVEL", logging.INFO))
 FORMAT = "%(asctime)-15s [%(levelname)8s] %(message)s"
 logging.basicConfig(format=FORMAT)
+
+
+def check_latest_version(raise_exception=False):
+    """
+    Check for the latest version of the library.
+    Parameters
+    ----------
+    raise_exception: :obj:`bool`
+        Raise a RuntimeError if a bad version is being used
+    """
+    import etelemetry
+    from pkg_resources import parse_version
+
+    logger = lgr
+
+    INIT_MSG = "Running {packname} version {version} (latest: {latest})".format
+
+    latest = {"version": "Unknown", "bad_versions": []}
+    result = None
+    try:
+        result = etelemetry.get_project("dandi/dandi-cli")
+    except Exception as e:
+        logger.warning("Could not check for version updates: \n%s", e)
+    finally:
+        if result:
+            latest.update(**result)
+            if parse_version(__version__) != parse_version(latest["version"]):
+                logger.info(
+                    INIT_MSG(
+                        packname="dandi", version=__version__, latest=latest["version"]
+                    )
+                )
+            if latest["bad_versions"] and any(
+                [
+                    parse_version(__version__) == parse_version(ver)
+                    for ver in latest["bad_versions"]
+                ]
+            ):
+                message = (
+                    "You are using a version of dandi cli with a critical "
```

Comment:
How about we provide specific info about how to upgrade, both in the bad versions case and in the !latest case?

e.g. `pip install --upgrade dandi`

Reply:
What if there is only downgrade to get away from the bad version? 
What if it is installed from Debian package? From conda? ...